packages/sdks/javascript/src/api/generated/models/virtual-host-relation.ts

```diff
@@ -27,29 +30,41 @@ export interface VirtualHostRelation {
      */
     id: number;
     /**
-     * The type of the Virtual Host (`name-based` or `ip-based`)
-     * @type {string}
+     * 
+     * @type {VirtualHostType}
      * @memberof VirtualHostRelation
      */
-    type: string;
+    type: VirtualHostType;
     /**
      * The name of the Virtual Host
      * @type {string}
      * @memberof VirtualHostRelation
      */
     name: string;
     /**
-     * The relative path to the root directory of the Virtual Host
+     * 
+     * @type {ApplicationRelation}
+     * @memberof VirtualHostRelation
+     */
+    application: ApplicationRelation | null;
+    /**
+     * The document root location, within the application directory, that contains the public files to be served when a user visits the domain name associated with this virtual host
      * @type {string}
      * @memberof VirtualHostRelation
      */
     root_directory: string;
     /**
-     * The address of the forward server
+     * The network name or IP address on which the application linked to this virtual host is configured to listen for incoming requests. A valid address has `http` or `https` protocol, a domain name or IP address, an optional port and optional path. You can also specify a UNIX-socket using `unix:` protocol. Examples: `http://127.0.0.1:8080` (for applications exposing port `8080`, for example running in a Docker container), `http://127.0.0.1:3000` (for applications kept alive by a daemon/background process that listens on port `3000`), `unix:/var/run/example.sock` (for applications listening on a custom socket)
      * @type {string}
      * @memberof VirtualHostRelation
      */
-    forward_server_address: string;
+    application_listen_address: string;
```

Comment:
Campo deveria ser do tipo  "string | null" semelhante ao "last_action" abaixo, pois usuário pode querer remover o campo ao editar e nem todas aplicação terá esse campo preenchido ao criar

Reply:
O campo já está marcado como "sometimes|nullable" nas requests. Atualizei no resource/collection e estou fazendo o build. Volto com noticias em alguns minutos.

Reply:
@souz4-my VirtualHostRelation vai ser o retorno do GET INDEX, então acredito que nesse caso ambos root_directory e application_listen_address precisariam ser no formato abaixo

```suggestion
    application_listen_address?: string;
```

É separado do problema descrito por ti, que pode ser encontrado em https://github.com/devopness/devopness/pull/955/files#diff-e6b8a9f9c59ebec763393ef7b81b0fa900207bd6ce46cf5b76efd3da0b981250R45

Reply:
@thlmenezes qual a diferença entre `field: string|null` e `field?: string`?

Reply:
Um GET em um Virtual Host sem vinculo com Application, devolve os campos `"application" => null` e `"application_listen_address" => null`.

Pelo que entendi pesquisando, a notação `field?: string` significa que o field `pode ou não existir`, e caso exista, ele tem o tipo string.

Neste caso, como sempre temos o field `application_listen_address`, acredito que o correto seja mesmo `string|null`.

![image](https://github.com/devopness/devopness/assets/15692310/1e6d64dd-6d1f-49b7-a5fc-aaabee95d68d)
![image](https://github.com/devopness/devopness/assets/15692310/c6954753-46d6-49f4-83d5-f6862460c593)

Reply:
> pode ou não existir, e caso exista, ele tem o tipo string

"pode ou não existir" == undefined
"pode ou não existir" != null

```ts
function formatFieldRequired(field: string | null){}

formatFieldRequired()
// error => An argument for 'field' was not provided
formatFieldRequired(undefined)
// error => Argument of type 'undefined' is not assignable to parameter of type 'string | null'
formatFieldRequired('') // ok
formatFieldRequired(null) // ok

function formatFieldNotRequired(field?: string){}

formatFieldNotRequired() // ok
formatFieldNotRequired(undefined) // ok
formatFieldNotRequired('') // ok
formatFieldNotRequired(null)
// error => Argument of type 'null' is not assignable to parameter of type 'string | undefined'

function formatFieldNotRequired2(field: string | undefined){}

formatFieldNotRequired2()
// error => An argument for 'field' was not provided
formatFieldNotRequired2(undefined) // ok
formatFieldNotRequired2('') // ok
formatFieldNotRequired2(null)
// error => Argument of type 'null' is not assignable to parameter of type 'string | undefined'
```

[TS Playground](https://www.typescriptlang.org/play/?#code/GYVwdgxgLglg9mABMOAnAtgQygMRgUwBsATAJXwEcQZV9iAKYAkgLkQGcpUYwBzRAD6IwIQoQCUAbwC+AKFkoM2PETKVqtBuNkB6HYnypUaRAF4AfIgCCSTKl4h0+MFGQmA5E1XvEAd0zswnCuAA7GAG4wxHQKaFi4zGpUNHT04NFMYHTaegZGJhbW9o7OrnDAiFAAniH4iO7p+Jl0PjCBYMGIAewwvGCYAEaEdVBwiCF2mE5QhojllTV17pzcfILCooTusUoJquTJmvTu7uKIuXAA1jvxKiQHGqkiYmcX1wrg0PBIireJAHLBB4pBheEgAfjYKx4vCkchuygBQPUIPor30VwRexIgKgwKOjWaxHRc2uv0Rqlx+NSJxJmPJ2OIVJRR2eEl0+kMxlQZksVmKThccwq1Vq9TZrXanW6vX6QxGYwmqCm+BmPPmoqW0LWQkJPBa8lAkFgCDcuzuTORhzoACZGIkoVwYes9VliTJDXEKTirY9iHacpz8jzCjYugLSmb6mDiD5-FLQhEojEGRbmdb-WkwBl9cTzhiyV7Gem-XbafnSVi076QXa2YG8tzeUUHIKyiLFuLNpKgq4ZX1BsNKorJtNZhrO8snTrEK6WkA)

Reply:
De toda forma, abri threads separadas pq aqui estamos discutindo mais de um tópico ao mesmo tempo